audiences/Rakefile

```
@@ -15,4 +16,24 @@ APP_RAKEFILE = File.expand_path("spec/dummy/Rakefile", __dir__)
 load "rails/tasks/engine.rake"
 load "rails/tasks/statistics.rake"
 
-task default: %i[rubocop spec]
+desc "Build audiences-react and copy JS to Rails assets"
+task build_frontend: :environment do
+  puts "Building audiences-react..."
+  system("cd ../audiences-react && yarn install && yarn build --mode ujs") || raise("Failed to build audiences-react")
+
+  source_path = "../audiences-react/dist/audiences-ujs.js"
+  destination_dir = "app/assets/builds"
+  destination_path = "#{destination_dir}/audiences-ujs.js"
+
+  FileUtils.mkdir_p(destination_dir)
+
+  raise "Build file not found: #{source_path}" unless File.exist?(source_path)
+
+  puts "Copying #{source_path} to #{destination_path}..."
+  FileUtils.cp(source_path, destination_path)
+  puts "Successfully copied!"
```

Did you consider using Pathname to clean this up and make this a bit for specific?

```suggestion
  source_path = Pathname.pwd.join("../audiences-react/dist/audiences-ujs.js")
  destination_dir = Pathname.pwd.join("app/assets/builds")
  destination_path = destination_dir.join("audiences-ujs.js")

  destination_dir.mkpath
  raise "Build file not found: #{source_path}" unless source_path.exist?

  puts "Copying #{source_path} to #{destination_path}..."
  source_path.copy(destination_path)
  puts "Successfully copied!"
```